### PR TITLE
Fix many issues in gdal algs (manual backport to 3.16)

### PR DESCRIPTION
**Pull request by agiudiceandrea**
## Description

Manual backport of cbcd505 (#41893 - #41847) to 3.16 branch.

I'm not completely sure the merge conflict in GdalUtils.py was correctly resolved.

### CODE REVIEW

File: python/plugins/processing/algs/gdal/GdalUtils.py [hunk 1]
```diff
@@ -217,13 +217,14 @@ def getFormatShortNameFromFilename(filename):
...
+        escChars = [' ', '&', '(', ')', '"', ';']
         joined = ''
         for s in strList:
             if not isinstance(s, str):
                 s = str(s)
-            if s and s[0] != '-' and any(c in s for c in escChars):
-                escaped = '"' + s.replace('\\', '\\\\').replace('"', '\\"') \
+            # don't escape if command starts with - and isn't a negative number, e.g. -9999
+            if s and re.match(r'^([^-]|-\d)', s) and any(c in s for c in escChars):
+                escaped = '"' + s.replace('\\', '\\\\').replace('"', '"""') \
```
Comment (agiudiceandrea) [inline python/plugins/processing/algs/gdal/GdalUtils.py:227]:
```suggestion
                escaped = '"' + s.replace('\\', '\\\\').replace('"', '\\"') \
```

**Review by nyalldawson — APPROVED** [review]

**Comment by nyalldawson** [comment]
@agiudiceandrea can you rebase and fix the conflicts please?

**Comment by agiudiceandrea** [comment]
@nyalldawson I followed the instructions at https://github.com/qgis/QGIS/pull/41893#issuecomment-787416583 and afterwards I (supposedly) followed the git and then GitHub advises to resolve rebase issue and merge conflicts.

I don't see now in GitHub merge conflicts to resolve. How can I spot them?
image

Anyway it could be less time consuming for you to close this PR and create a new one from scratch in the right way.

**Comment by nyalldawson** [comment]
> Anyway it could be less time consuming for you to close this PR and create a new one from scratch in the right way.

@agiudiceandrea don't read this the wrong way (it's not supposed to be aggresive!), but I'm not going to do that. If I do then it perpetuates the myth that I'm the only one who can do things the "right way" :laughing:

**Comment by agiudiceandrea** [comment]
Superseded by https://github.com/qgis/QGIS/pull/42408

**Pull request closed without merging**